File: audit_tarcker/static/java/auditor_page.js
document.addEventListener("DOMContentLoaded"  ,async function(){
        let name=document.getElementById('name');
        let tableBody=document.querySelector('.tbody');



        let response=await fetch("/auditor/auditor_details")
        let data=await response.json()


        auditorName=data['username']
        auditorId=data['id']
        alert(auditorId)
     // showing name -->
        name.innerText=auditorName;
        //getting auditor details from db
        async function fetchData(){
            try{
                response=await fetch("/auditor/data" ,{
                       method:'POST',
                       headers: {
                        "Content-Type": "application/json"
                    },
                    body :JSON.stringify({ "Id":auditorId ,'username':auditorName})
                })

                // checking if response is okay or not
                if (!response.ok) {
                    const errorText = await response.text();
                    throw new Error(`HTTP error! Status: ${response.status}, Message: ${errorText}`);
                }

                // getting response
                   data= await response.json()
                   console.log(data)

                   tableBody.innerHTML = ""; // Clear table before loading new data
                   data.forEach((item) => {
                        let row = document.createElement("tr");

                        // If in Update Mode, make 'audit_status' editable
                        row.innerHTML = `
                        <td class="bg-gray-800 text-white p-2 text-center">${item.Audit_id}</td>
                        <td class="bg-gray-800 text-white p-2 text-center">${item.auditor_id}</td>
                        <td class="bg-gray-800 text-white p-2 text-center">${item.auditor_name}</td>
                        <td class="bg-gray-800 text-white p-2 text-center ">${item.audit_type}</td>
                        <td class="bg-gray-800 text-white p-2 text-center">${item.planned_date}</td>
                        <td class="bg-gray-800 text-white p-2 text-center">${item.contact}</td>
                        <td class="bg-gray-800 text-white p-2 text-center">${item.email}</td>
                        <td class="bg-gray-800 text-white p-2 text-center">${item.client_id}</td>
                        <td class="bg-gray-800 text-white p-2 text-center">${item.state}</td>
                        <td class="bg-gray-800 text-white p-2 text-center">${item.location}</td>
                        <td class="bg-gray-800 text-white p-2 text-center">${item.payment_amount}</td>
                        <td class="paymentColor" >${item.payment_status}</td>
                        <td class="StatusColor"  >${item.audit_status}</td>
                    `;
                        tableBody.append(row);
                   })
            }catch(error){
                console.error("Error fetching audit data:", error);
            }
        }
        fetchData();
});

////securing updating status...
// async function checking(){
//    const submit=document.getElementById('submit');
//    const audit_id=document.getElementById('Id').value.trim();
//    const client_id=document.getElementById('clId').value.trim();
//    const select=document.getElementById('select');
//
//    let response=await fetch('https://finalavengers.onrender.com/auditor/check');
//    if(!response.ok){
//        console.log('the client was not matched ');
//    }
//    let data=await response.json();
//    console.log(data['message']);
//    if(response.ok){
//
//
//    }
//
//}
//
//
//
//// updating status details..

async function updateStatus(){

   const btn=document.getElementById('btnStatus');

   btn.addEventListener('click' ,async ()=>{
    const value=document.getElementById('update').value;

    let response1=await fetch("https://finalavengers.onrender.com/auditor/auditor_details")
        let data=await response1.json()

        auditorName=data['username']
        auditorId=data['id']

       const response=await fetch("https://finalavengers.onrender.com/auditor/status_update " ,{
                       method:'POST',
                       headers: {
                        "Content-Type": "application/json"
                             },
                        body :JSON.stringify({ "status": value ,"id":auditorId})
        })
            // checking if response is okay or not
        if (!response.ok) {
                    const errorText = await response.text();
                    throw new Error(`HTTP error! Status: ${response.status}, Message: ${errorText}`);
        }
        //getting response
        data= await response.json()
        alert(data['message'])
        showSuccessMessage(data['message'])
        function showSuccessMessage(message) {
            let msgDiv = document.getElementById("successMessage");
            msgDiv.innerText = message;
            msgDiv.style.display = "block";  // Show message
            msgDiv.style.opacity = "1";  // Reset opacity

            // Fade out effect after 3 seconds
            setTimeout(() => {
                let fadeEffect = setInterval(() => {
                    if (!msgDiv.style.opacity || msgDiv.style.opacity === "0") {
                        clearInterval(fadeEffect);
                        msgDiv.style.display = "none";  // Hide after fade out
                    } else {
                        msgDiv.style.opacity -= "0.1";  // Reduce opacity
                    }
                }, 100);
        }, 3000);
}


   })
}
updateStatus();

File: audit_tarcker/auditor_page.py
from flask import Flask ,redirect,render_template,Blueprint,request,jsonify,session,flash
import sqlite3
from flask_login import login_required,LoginManager,UserMixin,login_user
status=Blueprint('status' ,__name__)
import os
from audit_tarcker.config import get_connection


@status.route('/auditor/auditor_details')
def auditor_details():
    if session.get('role') == 'auditor':
        return jsonify({"username": session['username'], "id": session['id']})
    return jsonify({"error": "Unauthorized"}), 403

@status.route('/auditor/data',methods=["POST"])
@login_required
def status_update():
    try:
        data = request.get_json()
        print('this is received data', data)
        username = data['Id']
        password=data['username']
        print('this is username', username)
        query = f"SELECT * FROM audit_report WHERE auditor_id =%s "
        with get_connection() as conn:
            pointer = conn.cursor(dictionary=True)
            pointer.execute(query, (username,))
            row = pointer.fetchone()
            details=[]
            print(row)
            data1 = {'Audit_id': row['Audit_id'], 'auditor_name': row['auditor_name'],
                     'planned_date': row['planned_date'], 'state': row['State'],
                     'audit_status': row['Audit_status'], 'payment_amount': row['payment_amount'],
                     'payment_status': row['payment_status'], 'contact': row['Contact'],
                     'auditor_id': row['auditor_id'], 'audit_type': row['audit_type'],
                     'client_id': row['Client_id'], 'location': row['location'], 'email': row['email']}

        if data1:
            details.append(data1)
            return details
        else:
            return jsonify({"error": "No data found for this auditor"}),404
    except Exception as e:
        print(f"Error in status_update: {e}")
        return jsonify({"error": str(e)}), 500  # Always return a response


# auditor status update
@status.route('/auditor/status_update' ,methods=['POST'])
def update():
    try:
        data = request.get_json()
        audit_status = data['status']
        audit_id = data['id']
        print(audit_status)

        print('data received :', data)
        update = """update Audit_report set audit_status= %s where Audit_id=%s """
        with get_connection() as conn:
            pointer = conn.cursor(dictionary=True)
            pointer.execute(update, (audit_status, audit_id))
            pointer.execute(f'select * from audit_report  where Audit_id=?', (audit_id,))
            user = pointer.fetchone()
            print(f' updated at :{user}')
            return jsonify({"message": 'Status updated successfully....! Refresh the page...'})
    except Exception as e:
        print(e)
        return str(e)












